### ISSUE TITLE
Include the networkID when converting an address from str

### ISSUE THREAD

**Issue by neithanmo** [ChainSafe/forest#610]
**Issue summary**

When the string address is valid and contains a valid *network id* and protocol type [Protocol::ID](https://github.com/ChainSafe/forest/blob/main/vm/address/src/lib.rs#L186), the parsed address is being constructed using the default network id instead of the one included in the string being parsed. 
for example:

`let address = Address::from_str("f01")`

then, calling the `to_string()` method will give:

`t01` and not the initial `f01`  

which I think is not correct, unless there are some assumptions or restrictions I do not know about?
The network_id should be included, so something like this would be better?:

```
           if protocol == Protocol::ID {                                                               
               if raw.len() > 20 {                                                                     
                   // 20 is max u64 as string                                                          
                   return Err(Error::InvalidLength);                                                   
               }                                                                                       
               let id = raw.parse::<u64>()?;                                                           
               let mut address = Address::new_id(id);                                                  
               address.set_network(network);                                                           
               return Ok(address);                                                                         
             } 
```  

**Other information and links**

**Comment by austinabell** [comment]
ah, yeah, this was intentionally left out before because the network prefix was never serialized as bytes so there was never any need to store it (the prefix would still be lost whenever an address is serialized as bytes with the current protocol).

I was just leaving to see if they change to serialize the network prefix to bytes before I made this change, but I assume this might be final. 

I just have to think about what is the best way to set this up, since it would be equally as confusing to retain the prefix on stringifying but not on bytes serialization, with also the negative of keeping more data in memory with every address.

**Comment by neithanmo** [comment]
I understand It makes sense. 

> Would you just need a string encode function with a network parameter,

about the go-like encode function, no I do not need that, the current stringifying function includes it.

>  or is there a reason for storing this network prefix with the address?

actually, the network id is already part of the address:
```
#[derive(PartialEq, Eq, Clone, Debug, Hash, Copy)]                                                  
 pub struct Address {                                                                                
        network: Network,                                                                               
        payload: Payload,                                                                               
 }  
```
what we need is to be able to get the "exact" address from the string and compare them, although the id is not part of it when serialize to bytes, but it is part of the string being parsed. I see that the id is being included in the other cases where the protocol is different than ID, so I think to keep consistency, we might just include/update it when the protocol ID is used? does it make sense?

**Comment by austinabell** [comment]
> actually, the network id is already part of the address:

Ah, right, I completely forgot I added the network enum, was a while ago. Yeah, I'll update this

**Comment by austinabell** [comment]
> actually, the network id is already part of the address:

Honestly been so long since I made that change I had a brain lapse there. Fixed in #611

**Comment by neithanmo** [comment]
wow that was fast. Thank you !!
